src/script/microbit-interfacing/MicrobitSerial.ts

```diff
@@ -133,25 +141,43 @@ export class MicrobitSerial implements MicrobitConnection {
           buttons: true,
         });
         await this.usb.serialWrite(startCmd.message);
+        const periodicMessagePromise = new Promise<void>((resolve, reject) => {
+          onPeriodicMessageRecieved = resolve;
+          setTimeout(() => {
+            onPeriodicMessageRecieved = undefined;
+            reject(new Error('Failed to receive data from remote micro:bit'));
+          }, 500);
+        });
         await this.sendCmdWaitResponse(startCmd);
+        if (this.isReconnect) {
+          await periodicMessagePromise;
+        } else {
+          periodicMessagePromise.catch(async e => {
+            logError('Failed to initialise serial protocol', e);
+            await this.disconnectInternal(false, 'remote');
+            this.isConnecting = false;
```

Comment:
Can this happen after e.g. a manual click on disconnect and then connect? Does that matter?

Reply:
No I don't think so. As soon as we've established an initial connection, any subsequent connection is considered a reconnect so we won't go down this branch.